### PR TITLE
rbenv: fix head build and install manpage

### LINKED ISSUE #1474: rbenv's homebrew formula is broken for --HEAD installs
Opening this because I don't have the time yet to dig into it...

Installation of the homebrew formula using --HEAD currently fails:

```
$ HOMEBREW_NO_INSTALL_FROM_API=1 brew install rbenv --HEAD
==> Fetching dependencies for rbenv: ruby-build
==> Fetching ruby-build
==> Downloading https://ghcr.io/v2/homebrew/core/ruby-build/manifests/20230202
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/ruby-build/blobs/sha256:3dbf454935a5bd36e3f9cc64e1b8d1f3404d289123eda4877a9179474d2abeae
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:3dbf454935a5bd36e3f9cc64e1b8d1f3404d289123eda4877a9179474d2abeae?se=2023-02-03T19%3A0
######################################################################## 100.0%
==> Fetching rbenv
==> Cloning https://github.com/rbenv/rbenv.git
Cloning into '/Users/jasonkarns/.cache/Homebrew/rbenv--git'...
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
==> Installing dependencies for rbenv: ruby-build
==> Installing rbenv dependency: ruby-build
==> Pouring ruby-build--20230202.all.bottle.tar.gz
🍺  /opt/homebrew/Cellar/ruby-build/20230202: 588 files, 300.5KB
==> Installing rbenv --HEAD
Error: inreplace failed
libexec/rbenv:
  expected replacement of "\"${BASH_SOURCE%/*}\"/../libexec" with #<Pathname:/opt/homebrew/Cellar/rbenv/HEAD-61747c0/libexec>

READ THIS: https://docs.brew.sh/Troubleshooting

rbenv's formula was built from an unstable upstream --HEAD.
This build failure is expected behaviour.
Do not create issues about this on Homebrew's GitHub repositories.
Any opened issues will be immediately closed without response.
Do not ask for help from MacHomebrew on Twitter.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

I assume this project owns the changes to rbenv's formula.

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

fixes https://github.com/rbenv/rbenv/issues/1474

### CODE REVIEW

**Comment by chenrui333** [comment]
Still have the same test failure as seen in [ruby-build 20230208.1](https://github.com/Homebrew/homebrew-core/pull/122671)

File: Formula/rbenv.rb [hunk 1]
```diff
@@ -24,19 +24,26 @@ class Rbenv < Formula
 
   def install
     inreplace "libexec/rbenv" do |s|
-      s.gsub! '"${BASH_SOURCE%/*}"/../libexec', libexec
+      # TODO: The following line can be removed in the next release.
+      # https://github.com/rbenv/rbenv/pull/1428
```
Comment (carlocab) [inline Formula/rbenv.rb:28]: `brew audit` might not like this line.

File: Formula/rbenv.rb [hunk 2]
```diff
@@ -24,19 +24,26 @@ class Rbenv < Formula
 
   def install
     inreplace "libexec/rbenv" do |s|
-      s.gsub! '"${BASH_SOURCE%/*}"/../libexec', libexec
+      # TODO: The following line can be removed in the next release.
+      # rbenv/rbenv/pull/1428 (`brew audit` doesn't like URLs of merged PRs.)
```
Comment (carlocab) [inline Formula/rbenv.rb:28]: Hi @SMillerDev

Comment (chenrui333) [reply Formula/rbenv.rb:28]: yeah, it is kind of concern that I have.

**Comment by carlocab** [comment]
Ah. I think I know what the test failure is from. #122653

File: Formula/rbenv.rb [hunk 3]
```diff
@@ -50,15 +57,16 @@ def install
     foo_script.write "echo hello"
     chmod "+x", foo_script
 
-    # Test versions.
-    versions = shell_output("eval \"$(#{bin}/rbenv init -)\" && #{bin}/rbenv versions").split("\n")
+    # Test versions. The second `rbenv` call is a shell function; do not add a `bin` prefix.
```
Comment (chenrui333) [inline Formula/rbenv.rb:60]: Nice!! 👍

**Review by carlocab — APPROVED** [review]

**Comment by mislav** [comment]
Thanks @chenrui333 for the fixes!

**Pull request merged**